packages/core/tests/type-graph.test.js

```diff
@@ -1362,4 +1362,64 @@ describe("Issues", () => {
        },
     });
   });
+  test("Issue #113: Rest arguments don't work with generics 01", async () => {
+    const sourceAST = prepareAST(`
+      function a<T>(...items: Array<T>) {
```

Comment:
Would it be nice to have a test case for this:
```ts
function a<T, U>(...items: Array<T> | [T, U, T]) {
  return items
}

const arr01: Array<string> = a('str01', 'str03')
const arr02: Array<number> = a(2, false, 3)
const arr03: Array<boolean> = a(false, true, false, true)
```
?

Reply:
that does not type check

return type is `Array<T> | [T, U, T]` and that is not assignable to `Array<T>`

it won't work even without rest args [try](https://hegel.js.org/try#GYVwdgxgLglg9mABAQwDwBUA0iCqA+AChigFMBbAZwC5EBBAJ3uQE8M9EAfRAbS123QBdAJSIA3gChEieiSgh6SYuQoSAvhIkQEFKCkYAGAIw0GTVrvowwAc3YBeFAW4ByS8ZfY3UegYDMLiJaOnrIhgBMpowsqGAgZABGJPQOTtzh2MDIADYUJNh+Qdpguvq+flHmqAlwcNkkyGCpyM5ZufmIPiAdbXnYXSQiQA)

```js
function a<T, U>(items: Array<T> | [T, U, T]) {
  return items
}

const arr01: Array<string> = a(['str01', 'str03'])
const arr02: Array<number> = a([2, false, 3])
const arr03: Array<boolean> = a([false, true, false, true])
```